spec/integration/mentioning_spec.rb

```diff
@@ -23,73 +31,254 @@ def mention_stream_for(user)
   def users_connected?(user1, user2)
     user1.contacts.where(person_id: user2.person).count > 0
   end
-end
 
+  def post_status_message(user, aspects)
+    status_msg = nil
+    inlined_jobs do
+      post "/status_messages.json", status_message: {text: text_mentioning(user)}, aspect_ids: aspects
+      status_msg = StatusMessage.find(JSON.parse(response.body)["id"])
+    end
+    status_msg
+  end
+
+  def post_status_message_via_federation(text, recipient)
+    entity = FactoryGirl.build(
+      :status_message_entity,
+      author: remote_user_on_pod_b.diaspora_handle,
+      text:   text,
+      public: false
+    )
+
+    allow_callbacks(%i(queue_private_receive receive_entity fetch_related_entity))
+
+    xml = generate_xml(entity, remote_user_on_pod_b, recipient)
```

Comment:
You don't need this whole generate and parse xml here, just send the `entity` to the `:receive_entity` callback.